config/application.rb

```diff
@@ -37,6 +37,15 @@ class Application < Rails::Application
     )
     IdentityConfig.build_store(configuration)
 
+    unused_configuration_keys = configuration.keys - IdentityConfig.key_types.keys
+
+    if !unused_configuration_keys.empty?
+      warn <<-EOS.squish
+        WARNING: There are defined configuration keys that are not loaded by the application.
+        They are: #{unused_configuration_keys}
+      EOS
+    end
```

Comment:
is there a way we can log this in the prod-deployed boxes? Like `Rails.logger.warn` as well?

Reply:
Much to my disappointment, `Rails.logger` doesn't exist at this point unfortunately :/

Reply:
It will log, but only in provisioning logs, not any of the Rails logs (or if you start a console).

Reply:
somehow we have `config.active_job.logger` and `Identity::Hostdata.logger` 
what if we store the unused keys back on `IdentityConfig.unused_keys` or something and then add an initializer that logs them?

Reply:
That's a great idea, I'll give it a shot

Reply:
Updated in f62fbfb, much bigger fan of that approach